pkg/asset/installconfig/aws/validation.go

```diff
@@ -313,10 +324,24 @@ func validateServiceEndpoints(fldPath *field.Path, region string, services []aws
 	if err := utilerrors.NewAggregate(errs); err != nil {
 		allErrs = append(allErrs, field.Invalid(fldPath, services, err.Error()))
 	}
-
 	return allErrs
 }
 
+func validateRegion(region string) error {
+	ses, err := GetSessionWithOptions(func(sess *session.Options) {
+		sess.Config.Region = aws.String(region)
+	})
+	if err != nil {
+		return err
+	}
+	ec2Session := ec2.New(ses)
+	input := &ec2.DescribeRegionsInput{RegionNames: aws.StringSlice([]string{region})}
+	ctx, cancel := context.WithTimeout(context.TODO(), 25*time.Second)
+	defer cancel()
+	_, err = ec2Session.DescribeRegionsWithContext(ctx, input)
+	return err
+}
```

Comment:
I know that using the context has been discussed previously in this PR. But I cannot get over the seemingly arbitrary 25-second timeout. I still think it is better to pull-up the URL validation code from `validateServiceEndpoints` and re-use it here to validate accessibility of the endpoint URL.
```suggestion
	return validateEndpointAccessibility(ec2Session.Endpoint)
}

func validateEndpointAccessibility(endpointURL string) error {
	// Ignore e2e.local from unit tests.
	if endpointURL == "e2e.local" {
		return nil
	}
	URL, err := url.Parse(endpointURL)
	if err != nil {
		return err
	}
	port := URL.Port()
	if port == "" {
		port = "https"
	}
	conn, err := net.Dial("tcp", net.JoinHostPort(URL.Hostname(), port))
	if err != nil {
		return err
	}
	conn.Close()
	return nil
}
```